### PR TITLE
Documentation cleanup

### PR DESCRIPTION
**Pull request by KieranRatcliffeInvertedAI**
Performed some clean-up to update examples with changes in code, added documentation for large simulation functions, and improved readability.

### CODE REVIEW

**Comment by KieranRatcliffeInvertedAI** [comment]
One important point of discussion is whether to officially state in our documentation that "agent_attributes" is deprecated which I have done in this PR. What are your thoughts @Ruishenl @adscib @rf-ivtdai?

**Comment by Ruishenl** [comment]
> One important point of discussion is whether to officially state in our documentation that "agent_attributes" is deprecated which I have done in this PR. What are your thoughts @Ruishenl @adscib @rf-ivtdai?

For python client at least we can do it, but i probably would add a warning message if one attemps to use it.

**Comment by KieranRatcliffeInvertedAI** [comment]
> > One important point of discussion is whether to officially state in our documentation that "agent_attributes" is deprecated which I have done in this PR. What are your thoughts @Ruishenl @adscib @rf-ivtdai?
> 
> For python client at least we can do it, but i probably would add a warning message if one attemps to use it.

@Ruishenl I used the python package "warnings" so that it would force them to see the message instead of hoping the review the iai logger.

File: invertedai/api/drive.py [hunk 1]
```diff
@@ -149,6 +150,9 @@ def drive(
         )
         return response
```
Comment (Ruishenl) [inline invertedai/api/drive.py:152]: You can add `warnings.simplefilter("once")` so it won't keep printing the same message over and over

Comment (KieranRatcliffeInvertedAI) [reply invertedai/api/drive.py:152]: I tested this but I don't think it quite works with our stateless approach in drive.

Comment (KieranRatcliffeInvertedAI) [reply invertedai/api/drive.py:152]: I made it work by applying the filter into the __init__.py file.

**Review by Ruishenl — APPROVED** [review]

**Comment by KieranRatcliffeInvertedAI** [comment]
I made some updates to the documentation for our current offering (e.g. I mentioned we do support pedestrian agent types and that a programmatic map ingestion tool is in development). If these changes are okay then I will merge this PR.

**Comment by Ruishenl** [comment]
> I made some updates to the documentation for our current offering (e.g. I mentioned we do support pedestrian agent types and that a programmatic map ingestion tool is in development). If these changes are okay then I will merge this PR.

Sure, do you plan to update documentation branch now ?

**Comment by KieranRatcliffeInvertedAI** [comment]
> > I made some updates to the documentation for our current offering (e.g. I mentioned we do support pedestrian agent types and that a programmatic map ingestion tool is in development). If these changes are okay then I will merge this PR.
> 
> Sure, do you plan to update documentation branch now ?

Yes, which if I recall means this PR needs to get merged then develop needs to be merged in the "documentation" branch correct?

**Pull request merged**